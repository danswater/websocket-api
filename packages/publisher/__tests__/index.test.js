const { sign } = require('aws4');
const http = require('http');
const https = require('https');
const request = require('../request');
const createPublisher = require('../');

jest.mock('aws4');
jest.mock('../request');

const fakeEvent = {
  "requestContext": { 
    "routeKey": "$default", 
    "authorizer": "", 
    "messageId": "UgfzGcrpvHcCJyg=", 
    "integrationLatency": "", 
    "eventType": "MESSAGE", 
    "error": "", 
    "extendedRequestId": "UgfzGG6RvHcF8GA=", 
    "requestTime": "03/Feb/2019:05:18:37 +0000", 
    "messageDirection": "IN", 
    "stage": "dev", 
    "requestId": "UgfzGG6RvHcF8GA=", 
    "domainName": "70xqba8pp4.execute-api.us-west-2.amazonaws.com", 
    "connectionId": "UgfywcrkvHcCJyg=", 
    "apiId": "70xqba8pp4", "status": "" 
  }
};

beforeEach(() => {
  jest.resetAllMocks();
  sign.mockImplementation((args) => args);
  request.mockImplementation((method, args) => args);
});

test('dry run', async () => {
  const send = createPublisher(fakeEvent, { dry: true, port: 80 });
  
  await send('hi');

  expect(request).not.toHaveBeenCalled();
  expect(sign).toHaveBeenCalled();
});

test('port 80', async () => {
  const send = createPublisher(fakeEvent, { port: 80 });
  await send('hi');
  expect(request.mock.calls[0][0]).toEqual(http);
});

test('port 443', async () => {
  const send = createPublisher(fakeEvent, { port: 443 });
  await send('hi');
  expect(request.mock.calls[0][0]).toEqual(https);
});

test('buffer data', async () => {
  const send = createPublisher(fakeEvent, { port: 443 });
  const buf = new Buffer([0x88, 0x02]);
  await send(buf);
  expect(sign.mock.calls[0][0]).toHaveProperty('data', buf);
})