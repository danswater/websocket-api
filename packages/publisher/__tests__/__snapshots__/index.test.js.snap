// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`buffer data 1`] = `
Object {
  "body": Object {
    "data": Array [
      136,
      2,
    ],
    "type": "Buffer",
  },
  "headers": Object {
    "Content-Type": "application/octet-stream",
  },
  "host": "70xqba8pp4.execute-api.us-west-2.amazonaws.com",
  "method": "POST",
  "path": "/dev/%40connections/UgfywcrkvHcCJyg%3D",
}
`;

exports[`http 1`] = `
Object {
  "body": "\\"hi\\"",
  "headers": Object {
    "Content-Type": "application/json",
  },
  "host": "70xqba8pp4.execute-api.us-west-2.amazonaws.com",
  "method": "POST",
  "path": "/dev/%40connections/UgfywcrkvHcCJyg%3D",
}
`;

exports[`https 1`] = `
Object {
  "body": "\\"hi\\"",
  "headers": Object {
    "Content-Type": "application/json",
  },
  "host": "70xqba8pp4.execute-api.us-west-2.amazonaws.com",
  "method": "POST",
  "path": "/dev/%40connections/UgfywcrkvHcCJyg%3D",
}
`;
